ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -182,6 +188,26 @@ let find_and_update ~__context vgpu_type =
     end
   | _ -> fail ()
 
+(* since vgpu_type refs had been created in update, just need to get ref from vgpu_type.compatible* lists *)
+let update_compatible_lists ~__context vgpu_type vgpu_type_ref rc =
+  let get_type_from_model_name model =
+    let open Db_filter_types in
+    let expr = Eq(Field "model_name", Literal model) in
+    match Db.VGPU_type.get_internal_records_where ~__context ~expr with
+    | (_ref, _) :: _ -> _ref
+    | _ -> raise (Api_errors.Server_error(Api_errors.internal_error, ["Could not find vgpu_type according to model name" ]))
+  in
+  let compatible_types_in_vm =  List.map (fun model -> get_type_from_model_name model) vgpu_type.compatible_types_in_vm in
```

Comment:
The code 
```
let compatible_types_in_vm = if vgpu_type.compatible_types_in_vm = [] then [] else [vgpu_type] in
```
should work well in current situation, namely a VM only have one type of vGPU allowed, 
but in the future, the VM may have multiple types, then the list should not only contain itselfl, 
The value should be parsed from `vgpu_type.compatible_types_in_vm`

Reply:
Actually, Even for `create` function, the list should also be parsed from the sring list `vgpu_type.compatible_types_in_vm`,

But I think the problem here is, during the execution of `create` for one vGPU type, it may refer to other type of vGPU in its `compatible_types_in_vm`, which may have not been created at all, then We do not know its `Ref` value.

Reply:
Exactly, that is the problem in the create function. To solve that for the future, more general case, we need to have two rounds: first create all, then fill in the compatible lists. But we don't necessarily have to do that yet.

My main point is that we should do it the same for both the create and the update case (consistency).

Reply:
sure, if we do not have to do the two round ATM, it looks like the `create`  methon wins as it simpler, 
I will update the `udpate` function according.